src/net/IPv4Address.hxx

```diff
@@ -81,13 +81,9 @@ class IPv4Address {
 	static constexpr struct sockaddr_in Construct(struct in_addr address,
 						      uint16_t port) {
 		return {
-#if defined(__APPLE__)
-			sizeof(struct sockaddr_in),
-#endif
-			AF_INET,
-			ToBE16(port),
-			address,
-			{},
+			.sin_family = AF_INET,
```

Comment:
This is not portable. It is illegal C++ syntax, and works only because it uses a GCC extension.

Reply:
Oh, I'm sorry. I don't know much C++.
I didn't notice this feature was not yet standardized.

Then how would you fix this? Add a feature test macro?
I'd suggest removing the constexpr optimization altogether. This is no critical code path.